### DISCUSSION TITLE
Best Approach for Using a Master Instance to Manage Multiple Pocketbase Instances?

### DISCUSSION THREAD

**Discussion by suelitonx** [pocketbase/pocketbase#5852] [Q&A] [answered]
I would like to ask for help/tips on a **school management** application where I plan to use **multiple instances of Pocketbase**. The reason for having multiple instances is to isolate data for each school, making it easier to manage, maintain, backup, and handle problems and errors.

The initial idea is to have a **'master' instance**, where the login flow goes first, acting as an **intermediary to the other instances** and returning the server, user data, and authentication token of the isolated instance.

Well, can anyone give me tips on how to implement this? Or suggest another approach I might use?

Apologies for any spelling mistakes; I'm from Brazil.

![Screenshot_5](https://github.com/user-attachments/assets/79223dea-9df6-412e-97ff-96029f43d6d0)
![Screenshot_4](https://github.com/user-attachments/assets/68a7eefb-a9bc-4ffe-bb8c-12aa51995b8c)

**Comment by chnfyi** [comment]
Can you explain further why a 'master' instance as an intermediary is required in your project ?

**Comment by suelitonx** [comment]
@chnfyi Since the system will only be accessed by a single application, and since each instance has its own users, how do you know which instance to authenticate?

Note: each instance will have its own domain, for example:
- pb1.example.com
- pb2.example.com

My first idea is to create a route in the master instance that scans the "users" collection of all instances, and if the email is recognized in any of them, it tries to authenticate and returns the token, along with its respective domain.

**Reply by chnfyi** [reply, marked as answer]
From my understanding, you could achieve your idea _"scanning the users collection of all instances with user email"_ and using both `validatePassword` and `recordAuthResponse` into isolated instances routes that you request from the master instance to authenticate in those instances and retrieve data you need. But there may be performance and security considerations.

There might be two alternatives ways:
- Have all users in the master instance and sync users in isolated instance(s) they belong
- Use only isolated instances and make your users select the school they belong just before or during the login flow to authenticate to the right instance
